plugin/forward/health.go

```diff
@@ -1,17 +1,47 @@
 package forward
 
 import (
+	"crypto/tls"
 	"sync/atomic"
+	"time"
 
 	"github.com/miekg/dns"
 )
 
+// HealthChecker checks the upstream health.
+type HealthChecker interface {
+	Check(*Proxy) error
+	SetTLSConfig(*tls.Config)
+}
+
+type hc struct{ c *dns.Client }
```

Comment:
consider renaming hc to dnsHc, since we're going to have other kinds of hc in future. It would be nice that name to correspond the hc type